kubernetes/structures.go

```diff
@@ -635,13 +635,14 @@ func flattenLabelSelectorRequirementList(l []metav1.LabelSelectorRequirement) []
 }
 
 func flattenLocalObjectReferenceArray(in []api.LocalObjectReference) []interface{} {
-	att := make([]interface{}, len(in))
-	for i, v := range in {
-		m := map[string]interface{}{}
+	att := []interface{}{}
+	for _, v := range in {
 		if v.Name != "" {
-			m["name"] = v.Name
+			m := map[string]interface{}{
+				"name": v.Name,
+			}
+			att = append(att, m)
```

Comment:
So while this solves the crash in the original issue this PR closes, it actually creates a new problem where the next time you run `terraform plan` you're going to see a diff adding the `image_pull_secret` block again. This is because here we are omitting the block entirely from being set into the state so terraform thinks it needs to be created again. This will happen indefinitely. 

I had a look at what the Kubernetes API actually does here, and if we write an imagePullSecret with the name set to `""` it just ignores it and returns an empty object like this:

```yaml
imagePullSecrets:
- {}
```

So I think what we actually want to do here is remove the `if` statement, so we actually always return the map, and if no name is set then it's a map with the name set to `""`. I did that on this branch and it both fixes the crash, and doesn't cause the diff mentioned above.